resources/elastic_lb.rb

```diff
@@ -1,9 +1,17 @@
 actions :register, :deregister
 default_action :register
 state_attrs :aws_access_key,
-            :name
+            :elb_name
+
+identity_attr :elb_name
 
 attribute :aws_access_key,        kind_of: String
 attribute :aws_secret_access_key, kind_of: String
 attribute :aws_session_token,     kind_of: String, default: nil
-attribute :name,                  kind_of: String
+attribute :elb_name,              kind_of: String, name_attribute: true
```

Comment:
Please correct me if I'm wrong, but `name` is an internal field of the `Chef::Resource`. So when you override it, the string that goes into the name disappears.

For example:
```
aws_elastic_lb "Applying timeout changes to ELB" do
name "awseb-e-2-AWSEBLoa-9Z4D3SHF5FUY"
idle_timeout 300
end
```
will converge as "aws_elastic_lb[awseb-e-2-AWSEBLoa-9Z4D3SHF5FUY]" instead of "Applying timeout changes to ELB".

I haven't seen any other community cookbooks using `name` field for this sake. But I'm not really an experienced chef cookbook developer, so I might be wrong.